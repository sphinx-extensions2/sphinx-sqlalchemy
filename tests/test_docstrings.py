"""Basic tests"""
import os.path
import sys

from sphinx_pytest.plugin import CreateDoctree


def test_multiline_docstring_d212(sphinx_doctree_no_tr: CreateDoctree, snapshot):
    """Basic test for models with multiline docstring with D212 format

    see: https://docs.astral.sh/ruff/rules/multi-line-summary-first-line
    """
    sys.path.insert(0, os.path.join(os.path.dirname(__file__), "modules"))
    sphinx_doctree_no_tr.set_conf({"extensions": ["sphinx_sqlalchemy"]})
    result = sphinx_doctree_no_tr(
        ".. sqla-model:: module2.TestUserMultilineDocstringD212"
    )
    assert not result.warnings
    assert "\n".join([li.rstrip() for li in result.pformat().splitlines()]) == snapshot


def test_multiline_docstring_d213(sphinx_doctree_no_tr: CreateDoctree, snapshot):
    """Basic test for models with multiline docstring with D213 format

    see: https://docs.astral.sh/ruff/rules/multi-line-summary-second-line
    """
    sys.path.insert(0, os.path.join(os.path.dirname(__file__), "modules"))
    sphinx_doctree_no_tr.set_conf({"extensions": ["sphinx_sqlalchemy"]})
    result = sphinx_doctree_no_tr(
        ".. sqla-model:: module2.TestUserMultilineDocstringD213"
    )
    assert not result.warnings
    assert "\n".join([li.rstrip() for li in result.pformat().splitlines()]) == snapshot
