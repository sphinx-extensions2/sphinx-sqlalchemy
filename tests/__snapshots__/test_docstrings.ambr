# serializer version: 1
# name: test_multiline_docstring_d212
  '''
  <document source="<src>/index.rst">
      <definition_list classes="simple sqla">
          <definition_list_item>
              <term>
                  module2.TestUserMultilineDocstringD212
                   (
                  <emphasis>
                      dbusers_d212
                  )
              <definition>
                  <paragraph>
                      A
                      <literal>
                          user
                      .
                  <paragraph>
                      The user has a multi-line docstring:
                  <literal_block xml:space="preserve">
                      This is an indented code block
                  <paragraph>
                      This is the last line of the docstring.
                  <rubric>
                      Columns:
                  <table align="left" classes="colwidths-auto">
                      <tgroup cols="3">
                          <colspec>
                          <colspec>
                          <colspec>
                          <tbody>
                              <row>
                                  <entry>
                                      <paragraph>
                                          <emphasis>
                                              pk*
                                  <entry>
                                      <paragraph>
                                          INTEGER
                                  <entry>
                                      <paragraph>
                              <row>
                                  <entry>
                                      <paragraph>
                                          first_name
                                  <entry>
                                      <paragraph>
                                          VARCHAR?
                                  <entry>
                                      <paragraph>
                                          The name of the user.
                              <row>
                                  <entry>
                                      <paragraph>
                                          last_name
                                  <entry>
                                      <paragraph>
                                          VARCHAR(255)?
                                  <entry>
                                      <paragraph>
                                          The surname of the user.
                              <row>
                                  <entry>
                                      <paragraph>
                                          dob
                                  <entry>
                                      <paragraph>
                                          DATE
                                  <entry>
                                      <paragraph>
                                          The date of birth.
                  <rubric>
                      Constraints:
                  <bullet_list>
                      <list_item>
                          <paragraph>
                              PRIMARY KEY (pk)
                      <list_item>
                          <paragraph>
                              UNIQUE (first_name, last_name)
  '''
# ---
# name: test_multiline_docstring_d213
  '''
  <document source="<src>/index.rst">
      <definition_list classes="simple sqla">
          <definition_list_item>
              <term>
                  module2.TestUserMultilineDocstringD213
                   (
                  <emphasis>
                      dbusers_d213
                  )
              <definition>
                  <paragraph>
                      A
                      <literal>
                          user
                      .
                  <paragraph>
                      The user has a multi-line docstring:
                  <literal_block xml:space="preserve">
                      This is an indented code block
                  <paragraph>
                      This is the last line of the docstring.
                  <rubric>
                      Columns:
                  <table align="left" classes="colwidths-auto">
                      <tgroup cols="3">
                          <colspec>
                          <colspec>
                          <colspec>
                          <tbody>
                              <row>
                                  <entry>
                                      <paragraph>
                                          <emphasis>
                                              pk*
                                  <entry>
                                      <paragraph>
                                          INTEGER
                                  <entry>
                                      <paragraph>
                              <row>
                                  <entry>
                                      <paragraph>
                                          first_name
                                  <entry>
                                      <paragraph>
                                          VARCHAR?
                                  <entry>
                                      <paragraph>
                                          The name of the user.
                              <row>
                                  <entry>
                                      <paragraph>
                                          last_name
                                  <entry>
                                      <paragraph>
                                          VARCHAR(255)?
                                  <entry>
                                      <paragraph>
                                          The surname of the user.
                              <row>
                                  <entry>
                                      <paragraph>
                                          dob
                                  <entry>
                                      <paragraph>
                                          DATE
                                  <entry>
                                      <paragraph>
                                          The date of birth.
                  <rubric>
                      Constraints:
                  <bullet_list>
                      <list_item>
                          <paragraph>
                              PRIMARY KEY (pk)
                      <list_item>
                          <paragraph>
                              UNIQUE (first_name, last_name)
  '''
# ---
